pyro/distributions/torch_wrapper.py

```diff
@@ -37,8 +31,11 @@ def sample(self, sample_shape=torch.Size()):
             return self.torch_dist.sample(sample_shape)
 
     def batch_log_pdf(self, x):
-        batch_log_pdf_shape = self.batch_shape(x) + (1,)
+        sample_shape = broadcast_shape(self.batch_shape() + self.event_shape(), x.size(), strict=True)
```

Comment:
This is surprising, is it correct? I thought
```py
shape = sample_shape + batch_shape + event_shape
```
whereas this suggests that
```py
sample_shape >= shape = batch_shape + event_shape
```

Reply:
I think the variable name is confusing, will rename this to `shape` instead. This is basically the broadcasted shape of `x` and the distribution's batch + event shape.